### PR TITLE
add localize option to images

### PR DESCRIPTION
**Pull request by haroun**
## Summary

Add localize option to images

## What are the specific steps to test this change?

1. Login
2. Add an image using the image library
3. Edit the newly added image and select `Localize...` in the context menu
4. Export to another locale
5. Check the image on the new locale

apostrophecms-localize-image

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 1]
```diff
@@ -124,6 +125,12 @@ export default {
         return [];
       }
     },
+    isModified: {
+      type: Boolean,
+      default() {
+        return false;
+      }
```
Comment (ETLaurent) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:132]:
```suggestion
      default: false
```

since it's a primitive type we're setting, no need for a function (minor stuff 🥖 )

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 2]
```diff
@@ -152,11 +159,20 @@ export default {
     moduleOptions() {
       return window.apos.modules[this.activeMedia.type] || {};
     },
+    canLocalize() {
+      return (Object.keys(apos.i18n.locales).length > 1) && this.moduleOptions.localized && this.activeMedia._id;
```
[the remark this answers is not in the document]
Comment (ETLaurent) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:163]: I'm afraid that would be ok if we were checking `length > 0`, but here we're checking if there're several locales

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 3]
```diff
@@ -330,14 +346,35 @@ export default {
...
+        }
+      }
+      apos.bus.$emit('admin-menu-click', {
+        itemName: '@apostrophecms/i18n:localize',
+        props: {
+          doc: this.activeMedia
+        }
+      });
+    },
+    async close() {
```
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:376]: I can't find it either. I checked the `moreMenu` (which can invoke methods dynamically) and for possible calls in mixins.

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 4]
```diff
@@ -330,14 +346,35 @@ export default {
...
+      // the localize operation
+      if (this.isModified) {
+        if (!await this.confirmAndCancel()) {
+          return;
+        } else {
+          await this.cancel();
+          this.updateActiveDoc(this.activeMedia);
+        }
+      }
+      apos.bus.$emit('admin-menu-click', {
```
[the remark this answers is not in the document]
Comment (boutell) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:369]: The listener for that is implemented in `TheAposModals`. The logic there is hard to follow but it works.

[the remark this answers is not in the document]
Comment (ValJed) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:369]: Ok right, I re read the code and I get it, reassignments are a bit confusing. good to me 👍🏼

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (haroun) [reply modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:376]: it is called by the `AposModifiedMixin.js` cf. https://github.com/apostrophecms/apostrophe/blob/main/modules/%40apostrophecms/ui/ui/apos/mixins/AposModifiedMixin.js#L50

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 5]
```diff
@@ -330,14 +346,35 @@ export default {
...
+    async localize(media) {
+      // If there are changes warn the user before discarding them before
+      // the localize operation
+      if (this.isModified) {
+        if (!await this.confirmAndCancel()) {
+          return;
+        } else {
+          await this.cancel();
+          this.updateActiveDoc(this.activeMedia);
+        }
```
Comment (haroun) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:367]:
```suggestion
        }

        await this.cancel();
        this.updateActiveDoc(this.activeMedia);
```

File: modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue [hunk 6]
```diff
@@ -124,6 +125,10 @@ export default {
         return [];
       }
     },
+    isModified: {
+      type: Boolean,
+      default: false;
```
Comment (haroun) [inline modules/@apostrophecms/image/ui/apos/components/AposMediaManagerEditor.vue:130]:
```suggestion
      default: false
```

**Review by ETLaurent — APPROVED** [review]

**Review by ValJed — APPROVED** [review]
